### DISCUSSION TITLE
Webpack Dev Server memory grows in size and crashs sooner or later.

### DISCUSSION THREAD

**Discussion by SebastianStehle** [webpack/webpack#15847] [General]
Hi,

I have an issue with webpack dev server (4.7.4) and webpack (5.69.1).

It works fine but the memory usage grows with each change for about 100 MB and then eventually crashs when 6GB are reached. This is the maximum I have configured.

I have made a memory dump with Chrome and I see a lot of cashed files, but I don't know if this is normal or not.

Sometimes it is even the same file:
![image](https://user-images.githubusercontent.com/1236435/169283215-d58bf88c-ed8e-49e1-b2de-aeb7c1a5dcc3.png)

Any ideas how to debug this further?

**Comment by alexander-akait** [comment]
Hello, can your upload files here?

**Reply by SebastianStehle** [reply]
Which files? You mean my config or the dump?

**Reply by SebastianStehle** [reply]
I have uploaded the config here: https://gist.github.com/SebastianStehle/861557ec88da173a2080fd8970813363 ...

**Reply by alexander-akait** [reply]
@SebastianStehle dump, based on your screenshot `ts-loader` can ahve memory leaking, but need investigate

**Reply by SebastianStehle** [reply]
Here it is: https://www.dropbox.com/s/ok6h0xjjxqibhja/Heap-20220519T132551.heapsnapshot?dl=0

**Reply by alexander-akait** [reply]
Can you try to disable `ESLintPlugin`?

**Reply by SebastianStehle** [reply]
Does not help.

**Reply by SebastianStehle** [reply]
I also tried to disable StylelintPlugin

**Reply by vankop** [reply]
Could you provide your webpack config?

**Reply by vankop** [reply]
and could you try this without cache? ( `cache: false` in webpack.config )

**Reply by SebastianStehle** [reply]
Sorry, I have forgotten to add the link: https://gist.github.com/SebastianStehle/861557ec88da173a2080fd8970813363

**Reply by SebastianStehle** [reply]
Cache: false makes everything way slower but does not seem to have an impact.

**Reply by alexander-akait** [reply]
Sounds like we have a bug/limitation, can you try to set https://webpack.js.org/configuration/cache/#cachemaxgenerations, for large projects you need to have low value

**Reply by SebastianStehle** [reply]
It seems to help and the memory stays more or less constant. It slowly jump from 3.2GB to 4GB and then falls back to 3.2GB after some time. I guess this is the Node GC.

**Reply by alexander-akait** [reply]
@SebastianStehle Can you provide now many modules (and their size) you have? You can use webpack bundler analizer

**Reply by SebastianStehle** [reply]
https://www.dropbox.com/s/nysbxdypyxs5pc8/report.html?dl=0
